crates/tools/js/benchmark/index.js

```diff
@@ -0,0 +1,216 @@
+const path = require("path");
+const fs = require("fs");
+const readline = require("readline");
+const zlib = require("zlib");
+
+const { _ } = require("lodash");
+
+const {
+  createHardhatNetworkProvider,
+} = require("hardhat/internal/hardhat-network/provider/provider");
+
+const SCENARIOS_DIR = "../../scenarios/";
+
+async function main() {
+  const numArgs = process.argv.length;
+  let grep = undefined;
+  if (
+    (numArgs > 2 && process.argv[numArgs - 2] === "--grep") ||
+    process.argv[numArgs - 2] === "-g"
+  ) {
+    grep = process.argv[numArgs - 1];
+  }
+
+  const result = {};
+  const scenariosDir = path.join(__dirname, SCENARIOS_DIR);
+
+  // List files in scenarios directory
+  const scenarioFiles = await fs.promises.readdir(scenariosDir);
+  scenarioFiles.sort();
+  let totalTime = 0;
+  let totalFailures = 0;
+  for (let scenarioFile of scenarioFiles) {
+    if (grep && !scenarioFile.includes(grep)) {
+      continue;
+    }
+    // Get the filename from the path
+    const scenarioResult = await runScenario(
+      path.join(scenariosDir, scenarioFile)
+    );
+    totalTime += scenarioResult.result.timeMs;
+    totalFailures += scenarioResult.result.failures.length;
+    result[scenarioResult.name] = scenarioResult.result;
+  }
+
+  console.log(JSON.stringify(result));
+
+  // Log info to stderr so that it doesn't pollute stdout where we write the result
+  console.error(
+    `Total time ${
+      Math.round(100 * (totalTime / 1000)) / 100
+    } seconds with ${totalFailures} failures.`
+  );
+
+  process.exit(0);
+}
+
+async function runScenario(scenarioPath) {
+  const { config, requests } = await loadScenario(scenarioPath);
+  const name = path.basename(scenarioPath).split(".")[0];
+  console.error(`Running ${name} scenario`);
```

Comment:
Same question as above around usage of `console.error`

Reply:
https://github.com/NomicFoundation/hardhat/pull/4978#discussion_r1521278713